pkg/controller/daemon/update.go

```diff
@@ -570,3 +614,35 @@ func (h historiesByRevision) Swap(i, j int) { h[i], h[j] = h[j], h[i] }
 func (h historiesByRevision) Less(i, j int) bool {
 	return h[i].Revision < h[j].Revision
 }
+
+func (dsc *DaemonSetsController) checkShouldDeleteDaemonPod(nodeName string, ds *apps.DaemonSet) (bool, error) {
+
+	node, err := dsc.nodeLister.Get(nodeName)
+	if err != nil {
+		return false, err
+	}
+
+	shouldRun, shouldContinueRunning := NodeShouldRunDaemonPod(node, ds)
+
+	if !shouldRun || !shouldContinueRunning {
+		return true, nil
+	}
+	return false, nil
+
+}
+
+func getNumAvailablePod(ds *apps.DaemonSet, nodeToDaemonPods map[string][]*v1.Pod, now time.Time) int32 {
+
+	numAvailablePod := int32(0)
+
+	for _, pods := range nodeToDaemonPods {
+		for _, pod := range pods {
+
+			if podutil.IsPodReady(pod) && podutil.IsPodAvailable(pod, ds.Spec.MinReadySeconds, metav1.Time{Time: now}) && pod.ObjectMeta.DeletionTimestamp == nil {
```

Comment:
I think we are only interested in one pod per node when checking the availability (similar to status).

Reply:
Now that I've changed that logic to count availableNumber in the surge loop, I don't think I need to worry anymore.